amazon-kinesis-client/src/test/java/software/amazon/kinesis/application/TestConsumer.java

```diff
@@ -220,4 +258,53 @@ private void deleteResources(StreamExistenceManager streamExistenceManager, Leas
         log.info("-------------Finished deleting resources.----------------");
     }
 
+    @Data
+    @ToString
+    @AllArgsConstructor
+    public static class StreamScaler implements Runnable {
+        private final KinesisAsyncClient client;
+        private final String streamName;
+        private final Queue<ReshardOptions> scalingFactor;
+        private int totalRotations;
+        private KCLAppConfig testConfig;
+
+        @Override
+        public void run() {
+            try {
+                log.info("----------------------------Starting stream scale----------------------");
+                if (!scalingFactor.isEmpty()) {
+                    log.info("Starting stream scaling with params : {}", this);
+                    final DescribeStreamSummaryRequest describeStreamSummaryRequest = DescribeStreamSummaryRequest.builder().streamName(streamName).build();
```

Comment:
1. This does not support stream ARNs. Is stream ARN support coming in a future revision?
2. `streamName` isn't changing. If this is executed multiple times, this DSSR could be created once and reused.